google_shopping.rb

```diff
@@ -8,8 +8,120 @@
 google_data = JSON.parse(file)
 
 
-#outputs the full object
-puts google_data.inspect
+# #outputs the full object
+# puts google_data.inspect
 
-#outputs the first item's title
-puts google_data["items"][0]["product"]["title"]
\ No newline at end of file
+# #outputs the first item's title
+# puts google_data["items"][0]["product"]["title"]
+
+# Number 1, The kind of results you're are dealing are shopping#products. Go through the items and find all results that have kind of shopping#product. How many are there? Where else is this count information stored in the search results?
+puts "************************************************************************************************************"
+puts "Number 1 - Print the title of all products that have kind 'shopping#products *******************************"
+puts "************************************************************************************************************"
+puts
+items = google_data["items"]
+count_kinds = 0
+items.each do |items|
+    if items["kind"].index("shopping#product")
+        puts " • " + items["product"]["title"]
+        count_kinds += 1
+    end
+end
+puts
+puts "there are " + count_kinds.to_s + " kinds with shopping#products"
+puts "the number of kinds with shopper#products is also found at google_data['currentItemCount'] - " + google_data['currentItemCount'].to_s
+puts
+puts "************************************************************************************************************"
+
+#Number 2, Find all items with a backorder availability in inventories.
+count_backorder = 0
+puts "Number 2 - prints all stock with 'backorder' availability **************************************************"
+puts "************************************************************************************************************"
+puts
+items.each do |items|
+    availability = items["product"]["inventories"][0]["availability"]
+    title = items["product"]["title"]
+    if availability.index("backorder")
+        puts " • " + title + ", has availability - " + availability 
+        count_backorder += 1
+    end
+end
+puts
+puts "There is/are " + count_backorder.to_s + " orders on backorder"
+puts
+puts "************************************************************************************************************"
+
+#Number 3, Find all items with more than one image link.
+puts "Number 3 - Display items with more than one image link *****************************************************"
+puts "************************************************************************************************************"
+puts
+count_image = 0
+items.each do |items|
```

Comment:
For this iterator, I would suggest using ```|item|``` as your parameter name, since you are iterating over each item in items.